lib/multi_channel.ml

```diff
@@ -139,7 +139,7 @@ let rec recv_poll_loop mchan dls cur_offset =
 
 let recv_poll_with_dls mchan dls =
   try
-    Ws_deque.pop (Array.unsafe_get mchan.channels dls.id)
+    Ws_deque.steal (Array.unsafe_get mchan.channels dls.id)
```

Comment:
`Ws_deque.pop` returns the latest pushed item (like a stack https://github.com/ocaml-multicore/lockfree/pull/38 ), and as such doesn't allow `yield` to be run **after** we had a chance to go through the other tasks from the deque :/